cmd/cli/plugin/feature/activate.go

```diff
@@ -24,22 +30,112 @@ var FeatureActivateCmd = &cobra.Command{
 }
 
 func init() {
-	FeatureActivateCmd.Flags().StringVarP(&featuregate, "featuregate", "f", "tkg-system", "Activate a Feature gated by a particular FeatureGate")
+	FeatureActivateCmd.Flags().BoolVar(&userAllowsVoidingWarranty, "permanentlyVoidAllSupportGuarantees", false, "Allow for the permanent voiding of all support guarantees for this environment. For some features, e.g. experimental features, if a user sets the activation status to one that does not match the default activation, all support guarantees for this environment will be permanently voided.")
 }
 
 func featureActivate(cmd *cobra.Command, args []string) error {
 	featureName := args[0]
-	featureGateClient, err := featuregateclient.NewFeatureGateClient()
+
+	fgClient, err := featuregateclient.NewFeatureGateClient()
 	if err != nil {
-		return fmt.Errorf("couldn't get featureGateRunner: %w", err)
+		return fmt.Errorf("could not get FeatureGate client: %w", err)
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
 	defer cancel()
 
-	if err := featureGateClient.ActivateFeature(ctx, featureName, featuregate); err != nil {
-		return fmt.Errorf("couldn't activate feature %s: %w", featureName, err)
+	feature, err := fgClient.GetFeature(ctx, featureName)
+	if err != nil {
+		return fmt.Errorf("could not get Feature %s: %w", featureName, err)
+	}
+
+	gates, err := fgClient.GetFeatureGateList(ctx)
+	if err != nil {
+		return fmt.Errorf("could not get FeatureGate List: %w", err)
+	}
+
+	gateName, featRef := featuregateclient.FeatureRefFromGateList(gates, featureName)
+
+	var proceedWithVoidingWarranty bool
+	if willWarrantyBeVoided(featRef, feature) {
```

Comment:
where is the immutability check being done when activating the feature?

Reply:
It's done by the featuregate client in featuregates/client/pkg/featuregateclient/validation.go lines 51-61.

```bash
// featureActivationToggleAllowed checks if a Feature is considered immutable by its stability
// level and associated policy. Immutable means a Feature's activation setting cannot be toggled.
func featureActivationToggleAllowed(feature *corev1alpha2.Feature) error {
	stability := feature.Spec.Stability
	policy := corev1alpha2.GetPolicyForStabilityLevel(stability)

	if policy.Immutable {
		return fmt.Errorf("activation setting for Feature %s cannot be toggled as its stability level is %s: %w", feature.Name, stability, ErrTypeForbidden)
	}
	return nil
}
```